rotkehlchen/db/dbhandler.py

```diff
@@ -3316,13 +3347,16 @@ def update_web3_node(self, node: WeightedNode) -> None:
                 write_cursor=cursor,
                 proportion_to_share=ONE - node.weight,
                 exclude_identifier=node.identifier,
+                blockchain=node.node_info.blockchain,
             )
 
     def delete_web3_node(self, identifier: int) -> None:
         """Delete a web3 node based on name.
         May raise:
         - InputError if no entry with such name is in the database.
         """
+        node = self.get_web3_node_by_id(identifier)
```

Comment:
you would need to also change the docstring here. Since it now deletes by identifier and not name.